### DISCUSSION TITLE
Cannot run any poetry command involving python

### DISCUSSION THREAD

**Discussion by ElmoreV** [python-poetry/poetry#7696] [Q&A]
I don't know if this belongs here or in the issues page, but here goes.

I've been trying to find a way to get poetry to work in my environment, but I have been encountering more and more issues.

I had a previous almost empty conda environment called manim_poetry on which I installed using the online 
curl -sSL https://install.python-poetry.org | python - (python3 doesn't work for my case, somehow)
Using the installation guide, I could not get my git repository from being able to be imported as a module.
So I uninstalled it, just to start from scratch, using the same script, and deleted the conda environment so that no traces were left.

I created a new empty conda environment called test_poetry, installed python on it, ran the installer and ran 'poetry env use python' as per https://realpython.com/dependency-management-python-poetry/ to maybe solve the issues. However, now I get these errors.

```shell

(test_poetry) C:\Users\~name~\Git\manim>poetry shell

Command ['C:\\Users\\~name~\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\manim-5lIcB_8P-py3.11\\Scripts\\python.exe', '-I', '-W', 'ignore', '-'] errored with the following return code 103

Error output:
No Python at '"C:\Users\~name~\miniconda3\envs\manim_poetry\python.exe'


Input:
import sys

if hasattr(sys, "real_prefix"):
    print(sys.real_prefix)
elif hasattr(sys, "base_prefix"):
    print(sys.base_prefix)
else:
    print(sys.prefix)

```
And I'm basically: "Yes. That python executable doesn't exist anymore, that's right." However, I can't run "poetry env use /path/to/actual_python.exe" since it gives the same error. (It uses python to set the python interpreter?)

Even after re-installing poetry again and creating another conda environment, this keeps happening. I also completely recloned the git repository from its starting state, so it is probably not that the pyproject.toml is bad. Any hints on how to continue? Or where to find the config file, so I can manually change this?

**Comment by ElmoreV** [comment]
I eventually uninstalled poetry again in the test_poetry environment, tried pip uninstall poetry in the other environments, manually deleted all 'poetry' packages manually from "../../miniconda3/envs/manim_env/Lib/site-packages\" and "../../miniconda3/envs/manim_env/conda-meta\", "miniconda3\pkgs\" and "AppData\Local\pypoetry".
After that, the error was solved. It might have been that I thought I called one version of poetry, but accidentally called another install of poetry.

**Reply by principalideal0** [reply]
Poetry automatically uses, *even if you try every way possible to insist otherwise*, the version of Python it was installed with. If you are trying to poetry run something via a subprocess, it is impossible to force any other behavior. You can install pyenv, and poetry won't respect pyenv's selection *even* if you use the config flag that should modify this behavior. It is beyond aggravating.

I also had to uninstall, and then using the correct Python version for my project's pip, install pipx and then poetry so that the versions were properly locked to the right Python. If you try to just install pipx through Homebrew, you'll end up getting Python 3.12 bundled with, so you can't do that. Then it worked.

This all seems to stem from a design decision that is aging rather poorly for poetry to 'respect' existing environments.

**Reply by Anikavuk** [reply]
> Poetry автоматически использует, _даже если вы пытаетесь всеми возможными способами настаивать на обратном_, версию Python, с которой она была установлена.

я удалила python3.9, удалила poetry. установила Python 3.12.5, установила Poetry. И все равно выходит эта ошибка ```
Command ['C:\\Users\\HP\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\tennis-match-scoreboard-PSW9Nc7T-py3.9\\Scripts\\python.exe', '-I', '-W', 'ignore', '-c', 'import sys\n\nif hasattr(sys, "real_prefix"):\n    print(sys.real_prefix)\nelif hasattr(sys, "base_prefix"):\n    print(sys.base_prefix)\nelse:\n    print(sys.prefix)\n'] errored with the following return code 103

Error output:
No Python at 'C:\Users\HP\AppData\Local\Programs\Python\Python39\python.exe'
```

**Reply by mszegedy** [reply]
Holy *crap.* Thank you, principalideal0, this saved me an incredible amount of work. I could not have figured this out otherwise.
